node/service/src/lib.rs

```diff
@@ -1208,6 +1284,77 @@ where
 	})
 }
 
+/// Get the workers path by considering the passed-in `workers_path` option, or possible locations
+/// on the filesystem. See `new_full`.
+fn get_workers_paths(
+	given_workers_path: Option<std::path::PathBuf>,
+) -> Result<Vec<(std::path::PathBuf, std::path::PathBuf)>, Error> {
+	if let Some(path) = given_workers_path {
+		log::trace!("Using explicitly provided workers path {:?}", path);
+
+		if path.is_executable() {
+			return Ok(vec![(path.clone(), path)])
+		}
+
+		let mut prep_worker = path.clone();
+		let mut exec_worker = path.clone();
+		prep_worker.push(polkadot_node_core_pvf::PREPARE_BINARY_NAME);
+		exec_worker.push(polkadot_node_core_pvf::EXECUTE_BINARY_NAME);
+
+		// Check if both workers exist. Otherwise return an empty vector which results in an error.
+		return if prep_worker.exists() && exec_worker.exists() {
+			Ok(vec![(prep_worker, exec_worker)])
+		} else {
+			Ok(vec![])
```

Comment:
Why does this function supports to return an `Error` if you just return `Ok`?

Reply:
`?` is used in one place.